smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/CodegenVisitor.kt

```diff
@@ -29,8 +30,15 @@ import java.util.logging.Logger
 
 /**
  * Orchestrates Kotlin code generation
+ *
+ * @param context The [PluginContext] associated with the codegen run
+ * @param integrationCurator allows clients to make changes to integrations after
+ *  classpath discovery has occurred. Provided for testing purposes.
  */
-class CodegenVisitor(context: PluginContext) : ShapeVisitor.Default<Unit>() {
+class CodegenVisitor(
+    context: PluginContext,
+    integrationCurator: (List<KotlinIntegration>) -> List<KotlinIntegration> = { it }
```

Comment:
Needed in order to override Smithy's classloading service loader functionality, to control if a protocol generator is available or not from within a test.  If there is a cleaner way of doing this I'm all :ear: